.evergreen/download-mongodb.sh

```diff
@@ -695,18 +695,14 @@ download_and_extract_package ()
    MONGODB_DOWNLOAD_URL=$1
    EXTRACT=$2
 
-   # shellcheck disable=SC3028
-   SCRIPT_DIR=$(dirname ${BASH_SOURCE:-$0})
-   . $SCRIPT_DIR/handle-paths.sh
-
    if [ -n "${MONGODB_BINARIES:-}" ]; then
       cd "$(dirname "$(dirname "${MONGODB_BINARIES:?}")")"
    else
       cd $DRIVERS_TOOLS
    fi
 
    echo "Installing server binaries..."
-   "$SCRIPT_DIR/retry-with-backoff.sh" curl $MONGODB_DOWNLOAD_URL --output mongodb-binaries.tgz
+   curl $MONGODB_DOWNLOAD_URL --output mongodb-binaries.tgz
```

Comment:
We should be using curl_retry here. curl_retry is defined in this file.

Reply:
Oh you removed curl_retry in PYTHON-4734. How about we just revert that commit entirely (again)?